src/averageAges.js

```diff
@@ -55,7 +68,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const peopleWithMother = people.filter(
+    child => child.mother && people.find(
+      parent => parent.name === child.mother));
+
+  const filteredPeople = onlyWithSon ? peopleWithMother.filter(
+    child => child.sex === 'm') : peopleWithMother;
+
+  const ageDifferences = filteredPeople.map(
+    person => person.born - people.find(
+      parent => parent.name === person.mother).born);
```

Comment:
if you do this with find - you can reuse it if you move this to the separate func
```suggestion
  const peopleWithMother = people.filter(
    child => child.mother && getMother(people, child.name));

  const filteredPeople = onlyWithSon ? peopleWithMother.filter(
    child => child.sex === 'm') : peopleWithMother;

  const ageDifferences = filteredPeople.map(
    person => person.born - getMother(people, person.name).born);
```